docs-src/modules/developing/pages/specFiles.adoc

```diff
@@ -0,0 +1,241 @@
+:tabs-sync-option:
+
+= Spec Files 
+
+In addition to modifiying resources directly, you can also use a *spec file* to describe the desired state of Kaskada.
+
+A spec file is a YAML file describing a set of Kaskada resources, for examples tables and views.
+
+The `sync` command is used to update the state of your system to match the file. 
+
+[NOTE]
+====
+Currently `sync` is only supported in the CLI client.
+====
+
+== Spec File Format
+
```

Comment:
Maybe add the list of resources that the spec file understands, as a list or a small table of contents. Then each one gets a subsection.